client/shipping-label/views/index.js

```diff
@@ -1,112 +1,222 @@
 import { bindActionCreators } from 'redux';
 import Button from 'components/button';
+import CalypsoI18n from 'i18n-calypso';
 import { translate as __ } from 'lib/mixins/i18n';
+import LabelHistory from './history';
 import PurchaseLabelDialog from './purchase';
 import RefundDialog from './refund';
 import ReprintDialog from './reprint';
 import TrackingLink from './tracking-link';
 import Spinner from 'components/spinner';
+import Gridicon from 'components/gridicon';
+import Tooltip from 'components/tooltip';
 import formatDate from 'lib/utils/format-date';
 import * as ShippingLabelActions from 'shipping-label/state/actions';
 import notices from 'notices';
 import GlobalNotices from 'components/global-notices';
 import getFormErrors from 'shipping-label/state/selectors/errors';
 import canPurchase from 'shipping-label/state/selectors/can-purchase';
 import _ from 'lodash';
+import { sprintf } from 'sprintf-js';
 
-let needToFetchLabelsStatus = true;
+class ShippingLabelRootView extends Component {
+	constructor( props ) {
+		super( props );
 
-const ShippingLabelRootView = ( props ) => {
-	const renderPurchaseLabelFlow = () => {
+		this.renderLabel = this.renderLabel.bind( this );
+		this.openTooltip = this.openTooltip.bind( this );
+		this.closeTooltip = this.closeTooltip.bind( this );
+
+		this.needToFetchLabelsStatus = true;
+
+		this.state = {
+			showTooltips: props.shippingLabel.labels.map( () => false ),
+		};
+	}
+
+	componentDidMount() {
+		this.historyMetaBox = document.getElementById( 'wc-connect-shipping-label-history' );
+		this.renderHistory();
+	}
+
+	componentDidUpdate() {
+		this.renderHistory();
+	}
+
+	componentWillUnmount() {
+		ReactDom.unmountComponentAtNode( this.historyMetaBox );
+	}
+
+	renderHistory() {
+		ReactDom.render( <LabelHistory labelsHistory={ this.props.shippingLabel.labelsHistory } />, this.historyMetaBox );
+	}
+
+	openTooltip( index ) {
+		const showTooltips = [ ...this.state.showTooltips ];
+		showTooltips[ index ] = true;
+
+		this.setState( { showTooltips } );
+	}
+
+	closeTooltip( index ) {
+		const showTooltips = [ ...this.state.showTooltips ];
+		showTooltips[ index ] = false;
+
+		this.setState( { showTooltips } );
+	}
+
+	renderPurchaseLabelFlow() {
 		return (
-			<div>
+			<div className="wcc-metabox-label-item" >
 				<PurchaseLabelDialog
-					{ ...props.shippingLabel }
-					{ ...props } />
-				<Button onClick={ props.labelActions.openPrintingFlow } >
-					{ __( 'Create label' ) }
+					{ ...this.props.shippingLabel }
+					{ ...this.props } />
+				<Button onClick={ this.props.labelActions.openPrintingFlow } >
+					{ __( 'Create new label' ) }
 				</Button>
 			</div>
 		);
-	};
+	}
+
+	renderRefundLink( label ) {
+		const today = new Date();
+		const thirtyDaysAgo = new Date().setDate( today.getDate() - 30 );
+		if ( ( label.used_date && label.used_date < today.getTime() ) || ( label.created_date && label.created_date < thirtyDaysAgo ) ) {
+			return null;
+		}
```

Comment:
You're right. It makes sense for this to live in the Connect server. I opened Automattic/woocommerce-connect-server#711 and I think it could be handled in a separate PR.